validator/client/src/main/java/tech/pegasys/teku/validator/client/ValidatorStatusLogger.java

```diff
@@ -0,0 +1,135 @@
+ */
+
+package tech.pegasys.teku.validator.client;
+
+import static java.util.stream.Collectors.toList;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.concurrent.atomic.AtomicReference;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import tech.pegasys.teku.api.response.v1.beacon.ValidatorStatus;
+import tech.pegasys.teku.bls.BLSPublicKey;
+import tech.pegasys.teku.validator.api.ValidatorApiChannel;
+
+public class ValidatorStatusLogger {
+
+  private static final Logger LOG = LogManager.getLogger();
+
+  private static final int VALIDATOR_KEYS_PRINT_LIMIT = 20;
+
+  final Set<BLSPublicKey> validatorPublicKeys;
+  final ValidatorApiChannel validatorApiChannel;
+  final AtomicReference<Map<BLSPublicKey, ValidatorStatus>> latestValidatorStatuses =
+      new AtomicReference<>();
+
+  public ValidatorStatusLogger(
+      Set<BLSPublicKey> validatorPublicKeys, ValidatorApiChannel validatorApiChannel) {
+    this.validatorPublicKeys = validatorPublicKeys;
+    this.validatorApiChannel = validatorApiChannel;
+  }
+
+  public void printInitialValidatorStatuses() {
+    validatorApiChannel
+        .getValidatorStatuses(getAsIdentifiers(validatorPublicKeys))
+        .thenAccept(
+            maybeValidatorStatuses -> {
+              if (maybeValidatorStatuses.isEmpty()) {
+                LOG.error("Unable to retrieve validator statuses from BeaconNode.");
+                return;
+              }
+
+              Map<BLSPublicKey, ValidatorStatus> validatorStatuses = maybeValidatorStatuses.get();
+              if (validatorPublicKeys.size() < VALIDATOR_KEYS_PRINT_LIMIT) {
+                printValidatorStatusesOneByOne(validatorStatuses);
+              } else {
+                printValidatorStatusSummary(validatorStatuses);
+              }
+            })
+        .reportExceptions();
+  }
+
+  private void printValidatorStatusesOneByOne(
+      Map<BLSPublicKey, ValidatorStatus> validatorStatuses) {
+    for (BLSPublicKey publicKey : validatorPublicKeys) {
+      Optional<ValidatorStatus> maybeValidatorStatus =
+          Optional.ofNullable(validatorStatuses.get(publicKey));
+      maybeValidatorStatus.ifPresentOrElse(
+          validatorStatus ->
+              LOG.info(
+                  "Validator {} status is " + validatorStatus, publicKey.toAbbreviatedString()),
+          () -> LOG.error("Error retrieving status for validator {}", publicKey));
+    }
+  }
+
+  private void printValidatorStatusSummary(Map<BLSPublicKey, ValidatorStatus> validatorStatuses) {
+    Map<ValidatorStatus, AtomicInteger> validatorStatusCount = new HashMap<>();
+    for (ValidatorStatus status : validatorStatuses.values()) {
+      AtomicInteger count =
+          validatorStatusCount.computeIfAbsent(status, __ -> new AtomicInteger(0));
+      count.incrementAndGet();
+    }
```

Comment:
For symmetry with `printValidatorStatusesOneByOne`, should we also include an "unknown" status for validators that we fail to retrieve?  Or else log an extra message about statuses we failed to retrieve?

Reply:
Makes sense. Done.